hoomd/md/pytest/test_integrate.py

```diff
@@ -86,21 +89,27 @@ def test_validate_groups(simulation_factory, two_particle_snapshot_factory):
                                      methods=[nve1],
                                      integrate_rotational_dof=True)
     integrator.rigid = rigid
-
-    snapshot = two_particle_snapshot_factory(particle_types=['R', 'A'])
-    if snapshot.communicator.rank == 0:
-        snapshot.particles.body[:] = [0, 1]
-
     sim = simulation_factory(snapshot)
-    rigid.create_bodies(sim.state)
     sim.operations.integrator = integrator
 
     with pytest.raises(RuntimeError):
         sim.run(10)
 
+
+def test_overlapping_filters(simulation_factory, lattice_snapshot_factory):
+    snapshot = lattice_snapshot_factory()
+
+    integrator = hoomd.md.Integrator(dt=0, integrate_rotational_dof=True)
+
+    sim = simulation_factory(snapshot)
+    sim.operations.integrator = integrator
+
+    sim.run(0)
+
     nve1 = hoomd.md.methods.NVE(filter=hoomd.filter.Tags([0, 1]))
     nve2 = hoomd.md.methods.NVE(filter=hoomd.filter.Tags([0, 1]))
     integrator.methods = [nve1, nve2]
+    sim.run(0)
```

Comment:
Shouldn't this error as attaching will automatically run the validation?

Reply:
Nevermind I see now you call run above. I would want to see that errors too though to ensure it errors on attaching.

Reply:
It *only* errors on attaching, not when setting new methods. This second call to `run` passes because the `Integrator` is already attached.

Reply:
I thought it worth testing this behavior specifically, I'll add a comment. The 2nd test verifies that `validate_groups` throws errors on attach.